sdks/go/pkg/beam/util/gcsx/gcs_test.go

```diff
@@ -13,35 +13,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package gcsx_test
+package gcsx
 
 import (
-	"context"
-	"time"
+	"testing"
 
-	"cloud.google.com/go/storage"
-	"github.com/apache/beam/sdks/v2/go/pkg/beam/util/gcsx"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/internal/errors"
 )
 
-func Example() {
```

Comment:
The main argument for leaving it in, in the sidecar gcsx_test package: https://go.dev/blog/examples

Go will hoist examples like this into the package documentation, and they are executed as a part of the tests, causing a failure when they fail.

I agree this one isn't exactly the best example of it though (since it would end up actually trying to call the SDK).

But it still ensures that the example compiles, even if it's not validating the use.

https://pkg.go.dev/github.com/apache/beam/sdks/v2/go/pkg/beam/util/gcsx#example-package

Reply:
Huh, neat - go's auto-documentation is still a mystery to me 🙃 I added it back in (in its own file)